### ISSUE TITLE
[qwen-omni-utils] fetch_video can hang forever in the batched torchvision resize (OpenMP barrier) on many-core hosts — frame-sharded resize is bitwise-identical and immune

### ISSUE THREAD

**Issue by LukeLIN-web** [QwenLM/Qwen2.5-Omni#394]
### Summary

`fetch_video` can hang **permanently** (blocked, not spinning) in its final batched resize:

```
fetch_video (vision_process.py)
  -> transforms.functional.resize(video, [h, w], BICUBIC, antialias=True)
    -> torch.nn.functional.interpolate  (ATen CPU kernel, OpenMP-parallel)
```

We hit this twice in a long offline labeling run (~30k `fetch_video` calls over ordinary H.264 clips, decoded with the torchcodec backend). Each time the process froze forever inside that call — ~0% CPU (futex sleep, not a busy loop), no memory/VRAM pressure, a *different* ordinary clip each time (1280×720 and 540×360), and `faulthandler.dump_traceback_later` repeatedly printed the same innermost frame in `interpolate`. Each hang burned ~8 wall-clock hours before an external watchdog caught it.

### Environment

- qwen-omni-utils 0.0.9, torch 2.12.0+cu126, torchvision 0.27.0, torchcodec 0.14.0+cu126
- Linux, 96-core host (ATen intra-op = 48 OpenMP threads by default)

### Analysis

The batched resize is executed as one big ATen OpenMP parallel region — on a many-core host that is a 48+-thread `libgomp` team entered once per fetched clip. The two classic causes of libgomp barrier deadlocks do **not** apply here, which we verified:

- no fork-based multiprocessing in the process (cf. pytorch/pytorch#17199), and
- only a single `libgomp.so.1` mapped in `/proc/self/maps` (cf. pytorch/pytorch#98836).

What remains is a rare lost-wakeup on the OpenMP team barrier itself (same family as the pathologies in pytorch/pytorch#57715): threads park in futex wait inside the parallel region and never come back. It is far below qwen-omni-utils' code — but `fetch_video` chooses the execution shape that exposes it, and a several-hour inference/labeling job is exactly the workload that gets killed by a once-in-10⁴ hang.

### Workaround (verified bitwise-identical, no speed loss)

Resize per frame on plain Python threads with ATen intra-op threads clamped to 1. No OpenMP team is ever formed, so this class of hang is structurally impossible at this site; parallelism comes from the thread pool instead (ATen releases the GIL), and batch rows are computed independently, so the result is **bit-for-bit identical** to the batched call:

```python
from concurrent.futures import ThreadPoolExecutor
import torch
from torchvision.transforms import functional as F

def framewise_resize(video, size, **kw):          # video: (N, C, H, W)
    prev = torch.get_num_threads()
    torch.set_num_threads(1)
    try:
        with ThreadPoolExecutor(min(8, video.shape[0])) as pool:
            return torch.cat(list(pool.map(
                lambda i: F.resize(video[i:i+1], size, **kw).float(),
                range(video.shape[0]))))
    finally:
        torch.set_num_threads(prev)
```

Measured on the environment above, 32-frame 720p clip resized to 504×896 (BICUBIC, antialias):

| implementation | wall time | output |
|---|---|---|
| current batched resize (48-thread OMP) | 199 ms | reference |
| `torch.set_num_threads(1)` only | 921 ms | bitwise identical |
| **per-frame × 8 python threads (ATen 1-thread)** | **194 ms** | **bitwise identical** |

Bitwise parity also holds across both incident resolutions, odd target sizes, the upscale branch, and end-to-end through `fetch_video` on a real clip (patched vs unpatched process, `torch.equal` on the returned tensor).

### Suggestion

Consider frame-sharding the final resize in `fetch_video` (or offering it behind an env knob like the reader backend), so long-running many-core inference jobs are not exposed to whole-process hangs from a single OMP parallel region. Happy to send a PR if that is welcome.
